cicecore/cicedynB/infrastructure/ice_restart_driver.F90

```diff
@@ -524,57 +531,73 @@ subroutine restartfile (ice_ic)
       !-----------------------------------------------------------------
       ! ice mask for dynamics
       !-----------------------------------------------------------------
-      if (my_task == master_task) &
-           write(nu_diag,*) 'ice mask for dynamics'
-
-      call read_restart_field(nu_restart,0,work1,'ruf8', &
-           'iceumask',1,diag,field_loc_center, field_type_scalar)
-
-      iceUmask(:,:,:) = .false.
```

Comment:
What have you removed the default value?  If you are relying it being set to false by default, I think that's dangerous.  I'd keep these statements in the code.  The cost is tiny.

Reply:
I still think it should be set here independently.

If someone decide to remove or change the initialization elsewhere, the restart read is going off the rails.  No reason to have that.  The restart read should set the "entire field" cleanly and not rely on initialization elsewhere and then fill in a subset of values.